unitpackage/entry.py

```diff
@@ -573,6 +573,95 @@ def from_csv(cls, csvname, metadata=None, fields=None):
 
         return cls(package=package)
 
+    @classmethod
+    def _modify_fields(cls, original, alternative, keep_original_name_as=None):
+        r"""Updates in a list of fields (original) the field names with those
+        provided in a dictionary. The original name of the fields is kept with
+        the name `original` in the updated fields.
+
+        EXAMPLES::
+
+            >>> fields = [{'name': '<E>', 'unit':'mV'},{'name': 'I', 'unit':'mA'}]
+            >>> alt_fields = {'<E>':'E'}
+            >>> Entry._modify_fields(fields, alt_fields, keep_original_name_as='original')
+            [{'name': 'E', 'unit': 'mV', 'original': '<E>'}, {'name': 'I', 'unit': 'mA'}]
+
+        """
+        for field in original:
+            for key in alternative.keys():
+                if field["name"] == key:
+                    if keep_original_name_as:
+                        field.setdefault(keep_original_name_as, key)
+                    field["name"] = alternative[key]
+
+        return original
+
+    def rename_fields(self, field_names=None, keep_original_name_as=None):
```

Comment:
Why not have `field_names` as positional argument?
Are there any circumstances you would call this method without a dict of new field names?

Reply:
agree